dbt/adapters/spark/connections.py

```diff
@@ -3,94 +3,66 @@
 from dbt.adapters.base import Credentials
 from dbt.adapters.sql import SQLConnectionManager
 from dbt.logger import GLOBAL_LOGGER as logger
-import dbt.exceptions
 
 from TCLIService.ttypes import TOperationState as ThriftState
 from thrift.transport import THttpClient
 from pyhive import hive
 
+from hologram.helpers import StrEnum
+from dataclasses import dataclass
+from typing import Optional
+
+import dbt.exceptions
+import decimal
 import base64
 import time
 
 
-SPARK_CONNECTION_URL = "https://{host}:{port}/sql/protocolv1/o/0/{cluster}"
-
-SPARK_CREDENTIALS_CONTRACT = {
-    'type': 'object',
-    'additionalProperties': False,
-    'properties': {
-        'method': {
-            'enum': ['thrift', 'http'],
-        },
-        'host': {
-            'type': 'string'
-        },
-        'port': {
-            'type': 'integer',
-            'minimum': 0,
-            'maximum': 65535,
-        },
-        'user': {
-            'type': 'string'
-        },
-        'cluster': {
-            'type': 'string'
-        },
-        'database': {
-            'type': 'string',
-        },
-        'schema': {
-            'type': 'string',
-        },
-        'token': {
-            'type': 'string',
-        },
-        'connect_timeout': {
-            'type': 'integer',
-            'minimum': 0,
-            'maximum': 60,
-        },
-        'connect_retries': {
-            'type': 'integer',
-            'minimum': 0,
-            'maximum': 60,
-        }
-    },
-    'required': ['method', 'host', 'database', 'schema'],
-}
+class SparkConnectionMethod(StrEnum):
+    THRIFT = 'thrift'
+    HTTP = 'http'
 
 
+@dataclass
 class SparkCredentials(Credentials):
-    SCHEMA = SPARK_CREDENTIALS_CONTRACT
-
-    def __init__(self, *args, **kwargs):
-        kwargs.setdefault('database', kwargs.get('schema'))
-
-        super(SparkCredentials, self).__init__(*args, **kwargs)
+    host: str
+    method: SparkConnectionMethod
+    schema: str
+    cluster: Optional[str]
+    token: Optional[str]
+    user: Optional[str]
+    database: Optional[str]
+    port: int = 443
+    organization: str = '0'
+    connect_retries: int = 0
+    connect_timeout: int = 10
+
+    def __post_init__(self):
+        # spark classifies database and schema as the same thing
+        self.database = self.schema
 
     @property
     def type(self):
         return 'spark'
 
     def _connection_keys(self):
-        return ('host', 'port', 'cluster', 'schema')
+        return 'host', 'port', 'schema', 'organization'
```

Comment:
is cluster secret? If it's not, probably worth including it again so it shows up in `dbt debug` output.

Reply:
Cluster is just a random ID assigned by Spark/Databricks. I'll update this now.